sqlglot/optimizer/merge_subqueries.py

```diff
@@ -284,6 +293,17 @@ def _merge_order(outer_scope, inner_scope):
     outer_scope.expression.set("order", inner_scope.expression.args.get("order"))
 
 
+def _merge_hints(outer_scope, inner_scope):
+    inner_scope_hint = inner_scope.expression.args.get("hint")
+    if not inner_scope_hint:
+        return
+    outer_scope_hint = outer_scope.expression.args.get("hint")
+    if outer_scope_hint:
+        outer_scope_hint.args["expressions"].extend(inner_scope_hint.expressions)
```

Comment:
Patching expressions directly won't set `parent` on the child nodes.

```
outer_scope_hint.set("expressions", outer_scope_hint.expressions + inner_scope_hint.expressions)
```

Reply:
ah i didn't catch this, you should use

outer_scope.append("expressions", ...)

Reply:
ah woah, I didn't know we added `append`. Nice!

Reply:
We would need an `outer_scope.extend("expressions", ...)` right? I can add that if you want or I could also go with the set approach.

Reply:
Or write a for loop 😛

Reply:
just do a for loop append?

Reply:
lol fine